### ISSUE TITLE
Fix FP S4023 (`no-empty-interface`): Ignore when extending external interfaces

### ISSUE THREAD

**Issue by Bouke** [SonarSource/SonarJS#2089]
S4023 is derived from typescript-eslint/no-empty-interface. That rule has the ability to "silence warnings about extending a single interface without adding additional members". For example in the following code:

```ts
interface Array<T> extends Ember.ArrayPrototypeExtensions<T> { }
```

This should be allowed, or at least configurable.

See also: https://github.com/typescript-eslint/typescript-eslint/issues/1640

**Comment by Scoup** [comment]
This rule avoids using [declaration merge](https://www.typescriptlang.org/docs/handbook/declaration-merging.html) in some cases.
